cli/lsp/tsc.rs

```diff
@@ -232,7 +232,9 @@ impl TsServer {
             started = true;
           }
           let value = request(&mut ts_runtime, state_snapshot, req, token);
-          if tx.send(value).is_err() {
+          // Don't print the send error if the token is cancelled, it's expected
+          // to fail in that case and this commonly occurs.
+          if tx.send(value).is_err() && !token.is_cancelled() {
```

Comment:
I think that doesn't work for the explicit cancels we do for e.g. getting diagnostics. A response is still expected in that case since the executing future isn't dropped i.e. no bail out on the await point.

Reply:
The child token is the one passed in to requests so it works, I don't see this message anymore.

I think a comment about that would just confuse, conversely the dropped-future-bail-out stuff is the obscure part and the default assumption would be that a response is necessary.

Reply:
Good point

Reply:
Ok, can you move the action out of the condition to make it more clear that the action needs to be done regardless of the other condition? Actions in an if statement with other conditions are kind of sketchy.

Reply:
Done, I was worried about the linter but I just ignored the error